File: C/Assignment10/Program1.c
//////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                              //
// File name :   Program1.c                                                                     //
// Description : Program which accept number from user & return the count of Even Digits.       //
// Author :      Pallavi Omprakash Malewar                                                      //
// Date :        20/06/2025                                                                     //
//                                                                                              //
//////////////////////////////////////////////////////////////////////////////////////////////////

/*
Input : 2395
Output : 1

Input : 1018
Output : 2

Input : -1018
Output : 2

Input : 8462
Output : 4
*/

#include<stdio.h>

//////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                              //
// Description   : Function accepts integer and returns count of even digits present in it.     //
// Function Name : CountEven                                                                    //
// Input         : int iNo                                                                      //
// Output        : int (Count of even digits)                                                   //
// Author        : Pallavi Omprakash Malewar                                                    //
// Date          : 20/06/2025                                                                   //
//                                                                                              //
//////////////////////////////////////////////////////////////////////////////////////////////////

int CountEven(int iNo)
{
    int iDigit = 0 , iCnt = 0;

    while(iNo != 0)
    {
        iDigit = iNo % 10;
        if((iDigit % 2) == 0)
        {
            iCnt++;
        }
        iNo = iNo / 10;
    }
    return iCnt;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter Number : ");
    scanf("%d",&iValue);

    iRet = CountEven(iValue);

    printf("Count of present Even Digits : %d\n",iRet);

    return 0;
}

File: JAVA/Assignment10/Program1.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write a program which accept number from user and return the    //
//                     count of even digits.                                           //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 30 Jun 2026                                                                  //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////

/*

    Input: 2395
    Output : 1

    Input: 1018
    Output : 2

    Input: -1018
    Output : 2

    Input: 8462
    Output : 4

*/


import java.util.Scanner;

class Program1
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter the Number : ");
        int No = sobj.nextInt();

        int Count = CountEven(No);

        System.out.println(Count);
    }

    public static int CountEven(int No)
    {
        if(No == 0) return 1;
        int iCount = 0;

        while(No != 0)
        {
            if((No % 10) % 2 == 0)
            {
                iCount++;
            }
            No /= 10;
        }
        return iCount;
    }
}